commit 6b1c44a20707fa9c44018fd82ee063be269a992c
Author: John Levon <john.levon@joyent.com>
Date:   2019-08-30T10:42:52+00:00 (6 weeks ago)
    
    OS-7973 fbt entry probes incorrectly placed due to -fshrink-wrap

diff --git a/usr/src/uts/Makefile.uts b/usr/src/uts/Makefile.uts
index 1256d73d5d..e95700780a 100644
--- a/usr/src/uts/Makefile.uts
+++ b/usr/src/uts/Makefile.uts
@@ -24,7 +24,6 @@
 # Copyright (c) 2011 Bayard G. Bell. All rights reserved.
 # Copyright (c) 2011,2017 by Delphix. All rights reserved.
 # Copyright (c) 2013 Andrew Stormont.  All rights reserved.
-# Copyright 2018 Joyent, Inc.
 # Copyright 2016 Hans Rosenfeld <rosenfeld@grumpf.hope-2000.org>
 # Copyright 2019 Joyent, Inc.
 #
@@ -251,6 +250,12 @@ CFLAGS_uts_i386 += -_gcc7=-mindirect-branch-register
 CFLAGS_uts_i386 += -_gcc8=-mindirect-branch=thunk-extern
 CFLAGS_uts_i386 += -_gcc8=-mindirect-branch-register
 
+#
+# Ensure that the standard function prologue remains at the very start of a function,
+# so DTrace fbt will instrument the right place.
+#
+CFLAGS_uts_i386 = -_gcc=-fno-shrink-wrap
+
 CSTD = $(CSTD_GNU99)
 
 CFLAGS_uts		=
